### ISSUE TITLE
Build loss-weight tuning script for landmark training

### ISSUE THREAD

**Issue by argiacomi** [argiacomi/AccurateFacialLandmarkDetection#7]
## Context

Add an experiment orchestration script that tunes the landmark training loss weights and outputs the recommended weighting configuration to use for training.

The script should cover the staged tuning plan:

1. Baseline run
2. Small manual STAR bracket
3. Optuna narrow search over STAR + consistency + auxiliary
4. Rerun top configs with multiple seeds
5. Pick by heldout 68 NME + hard-slice performance

### 1. Add a loss-weight tuning entrypoint

Create a script such as:

```text
tools/landmarks/tune_loss_weights.py
```

Acceptance criteria:

- Can launch training/evaluation jobs with different loss-weight configs.
- Supports dry-run mode that prints commands without executing them.
- Supports resuming from an existing tuning study/output directory.
- Stores all configs, metrics, command lines, seeds, and artifacts in a reproducible output directory.

### 2. Run and record the baseline configuration

The script should first run the locked baseline configuration.

Recommended baseline:

```text
star_loss_weight = 0.0
schema_consistency_weight = current default, e.g. 0.05
auxiliary_loss_weight = current default or configured baseline, e.g. 0.05 / 0.1
locw = fixed
hw = fixed
```

Acceptance criteria:

- Baseline metrics are stored separately from tuning trials.
- Baseline config is written to JSON/YAML.
- Later trials report deltas versus baseline.

### 3. Run a small manual STAR bracket

Before Optuna, run a small fixed STARLoss_v2 bracket to determine whether STAR is useful and identify a safe search range.

Recommended bracket:

```text
star_loss_weight = 0.0
star_loss_weight = 0.005
star_loss_weight = 0.01
star_loss_weight = 0.02
star_loss_weight = 0.05
```

Acceptance criteria:

- Consistency and auxiliary weights remain fixed during this bracket.
- Results are ranked by heldout 68 NME and hard-slice metrics.
- The script can optionally shrink the Optuna STAR search range based on bracket results.

### 4. Run Optuna narrow search

Use Optuna to tune only the narrow set of loss weights:

```text
star_loss_weight
schema_consistency_weight
auxiliary_loss_weight
```

Initial search ranges:

```text
star_loss_weight:          0.0 – 0.03
schema_consistency_weight: 0.0 – 0.08
auxiliary_loss_weight:     0.0 – 0.1
```

Acceptance criteria:

- Supports configurable number of trials.
- Supports parallel workers if available.
- Supports pruning only after a configurable minimum epoch / minimum validation signal.
- Does not tune unrelated parameters such as learning rate, batch size, sampler ratios, or architecture by default.
- Optuna study database/path is persisted.

### 5. Optimize for heldout 68 NME plus hard-slice performance

The objective should not optimize only overall NME. It should account for production 68 performance and hard slices.

Recommended objective shape:

```text
score =
  heldout_68_nme
+ hard_slice_weight * profile_nme
+ hard_slice_weight * occlusion_nme
+ hard_slice_weight * blur_or_lowres_nme
+ regression_penalty_if_frontal_or_overall_gets_worse
```

Acceptance criteria:

- Objective is configurable from CLI/config.
- Required slices include at least profile, occlusion, profile+occlusion if available, blur/low-quality if available, and overall 68.
- Missing slices are handled explicitly and reported.
- Selection prioritizes heldout 68 NME and prevents easy/frontal regression.

### 6. Rerun top configs with multiple seeds

After Optuna, rerun the top configs with multiple seeds to avoid selecting a noisy winner.

Acceptance criteria:

- Configurable `top_k` and seed list.
- Reports mean, standard deviation, and best/worst metrics for each finalist.
- Final selection is based on aggregate multi-seed performance.

### 7. Output the recommended training weights

The script should produce a final recommendation artifact.

Acceptance criteria:

- Writes `best_loss_weights.json` or similar.
- Includes recommended values for:
  - `star_loss_weight`
  - `schema_consistency_weight`
  - `auxiliary_loss_weight`
  - fixed `locw`
  - fixed `hw`
- Includes rationale: baseline delta, hard-slice delta, multi-seed stability, and any regressions.
- Prints the exact training flags to use for the selected configuration.

### 8. Add tests and documentation

Acceptance criteria:

- Unit tests cover objective scoring, missing-slice handling, config generation, top-k finalist selection, and final recommendation output.
- Dry-run test verifies generated commands without launching real training.
- README/docs explain how to run baseline, bracket, Optuna search, finalist reruns, and how to interpret the output.

## Definition of done

This is complete when the repo has a reproducible tuning script that can run baseline, STAR bracket, Optuna search, multi-seed finalist reruns, and emit the recommended loss weights for training based on heldout 68 NME plus hard-slice performance.
